File: FootScout-Vue.WebClient/src/models/interfaces/PlayerAdvertisement.ts
import type { PlayerPosition } from './PlayerPosition';
import type { PlayerFoot } from './PlayerFoot';
import type { SalaryRange } from "./SalaryRange";
import type { UserDTO } from '../dtos/UserDTO';

// Reprezentuje ogłoszenie piłkarza, zawierając jego pozycję, wiek, wzrost, preferowaną stopę, 
// zakres wynagrodzenia i daty ważności ogłoszenia.

export interface PlayerAdvertisement {
    id: number;
    playerPositionId: number;
    playerPosition: PlayerPosition;
    league: string;
    region: string;
    age: number;
    height: number;
    playerFootId: number;
    playerFoot: PlayerFoot;
    salaryRangeId: number;
    salaryRange: SalaryRange;
    creationDate: string;
    endDate: string;
    playerId: string;
    player: UserDTO;
}

File: FootScout-Vue.WebAPI/Entities/PlayerAdvertisement.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace FootScout_Vue.WebAPI.Entities
{
    // Model (encja) ogłoszenia piłkarskiego
    public class PlayerAdvertisement
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int PlayerPositionId { get; set; }

        [ForeignKey("PlayerPositionId")]
        public virtual PlayerPosition PlayerPosition { get; set; }

        [Required]
        [MaxLength(30)]
        public string League { get; set; }

        [Required]
        [MaxLength(30)]
        public string Region { get; set; }

        [Required]
        public int Age { get; set; }

        [Required]
        public int Height { get; set; }

        [Required]
        public int PlayerFootId { get; set; }

        [ForeignKey("PlayerFootId")]
        public virtual PlayerFoot PlayerFoot { get; set; }

        [Required]
        public int SalaryRangeId { get; set; }

        [ForeignKey("SalaryRangeId")]
        public virtual SalaryRange SalaryRange { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime CreationDate { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime EndDate { get; set; }

        [Required]
        public string PlayerId { get; set; }

        [ForeignKey("PlayerId")]
        public virtual User Player { get; set; }
    }
}